eureka2-testkit/src/test/java/com/netflix/eureka2/testkit/embedded/server/EmbeddedWriteServerTest.java

```diff
@@ -32,7 +34,9 @@ public void testRegistrationAndDiscoveryServices() throws Exception {
                 .build();
 
         InstanceInfo instanceInfo = SampleInstanceInfo.DiscoveryServer.build();
-        eurekaClient.register(instanceInfo).toBlocking().lastOrDefault(null);
+        RegistrationRequest request = eurekaClient.connect(Observable.just(instanceInfo));
+        request.subscribe();
+        request.getInitObservable().toBlocking().lastOrDefault(null);
```

Comment:
This looks like repetitive two statements. We usually care about first registration and ignore lifecycle. It might be more handy if RegistrationRequest observable is for first registration, while there is registrationRequest.lifecycle() if somebody cares about it.